compiler/src/iree/compiler/Codegen/Common/LowerMicroKernelsToCalls.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Interfaces/MicroKernelOpInterface.h"
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+struct LowerMicroKernelOpsToCallsPass
+    : LowerMicroKernelOpsToCallsBase<LowerMicroKernelOpsToCallsPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<memref::MemRefDialect>();
+  }
+  void runOnOperation() override;
+};
+}  // namespace
+
+void LowerMicroKernelOpsToCallsPass::runOnOperation() {
+  MLIRContext *context = &getContext();
+  RewritePatternSet patterns(context);
+  SmallVector<Operation *> toDelete;
+  Operation *errorOp = nullptr;
+  IRRewriter rewriter(context);
+  WalkResult result = getOperation().walk(
```

Comment:
(Asking for my own edification) - what is the motivation for the custom `Walk` here as opposed to a typical `OpRewritePattern` - is it that you want to match not just one specific Op class but all ops implementing `IREE::Codegen::MicroKernelOpInterface` and there is no ready made `RewritePattern` subclass for this case?

Reply:
Using `OpRewritePattern` and `applyPatternsAndFoldGreedily` allows for patterns to fail, i.e. a pattern might fail to apply but the latter method does not flag that as an error. As a result, the pass might just do nothing when the micro kernel operation does not lower to a function call. In this case, that should probably be flagged as a hard error. Thats why using a walk here.